File: hyeon-uk/2021/1222~0102/7347.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args){
        Scanner sc=new Scanner(System.in);
        int t=sc.nextInt();

        while((t--)>0){
            int even=0;
            int odd=0;
            int n=sc.nextInt();
            for(int i=0;i<n;i++){
                int temp=sc.nextInt();
                if(i%2==0&&temp==1){
                    even++;
                }
                else if(i%2==1&&temp==1){
                    odd++;
                }
            }
            if(n%2==1 || (n%2==0 && Math.abs(even-odd)<=1)){
                System.out.println("YES");
            }
            else{
                System.out.println("NO");
            }
        }
    }
}

File: jochanmin/2022/20220101/7347.cpp
#include<iostream>
using namespace std;
int arr[30];

int solve(int n){
    if(n%2==1) return 1;
    int odd=0,even=0;
    for(int i=0;i<n;i++){
        if(arr[i]==1){
            if(i%2==1) odd++;
            else even++;
        }
    }
    if(-1<= (even-odd) && (even-odd)<=1) return 1;
    return 0;
}
int main(){
    int t,n;
    cin>>t;
    while(t--){
        cin>>n;
        for(int i=0;i<n;i++) cin>>arr[i];
        if(solve(n)==1) cout<<"YES"<<endl;
        else cout<<"NO"<<endl;
    }
    return 0;
}